Scripts: Exile's Reach: Enhanced Combat Tactics and Northbound

malcrom opened the pull request:
All classes are now working like offical for combat tactics quest.
Northbound quest also complete.

Commit:
Scripts: Exile's Reach: Enhanced Combat Training WIP

Working on 5th class Shaman

Commit:
Update zone_exiles_reach.cpp

Commit:
Update 9999_99_99_99_world.sql

Commit:
Update Dec 10

SQL:
Conversation data complete
Conversation conditions complete
CODE:
Some cleanup.
Set start text for all classes.
Completed Monk quest.

Current todo: Shaman, Mage, Warlock, & Druid.

Commit:
Dec 10 2

Updated conversation builder to use GUIDs set on summon rather than checking team every time.

malcrom commented:
Can't get spell 320741 to create areatrigger.
Error message : AreaTrigger (areaTriggerCreatePropertiesId 19479) not created. Invalid areatrigger create properties id (19479).

Commit:
Dec 11 Update

Added proper spell scrips.
Added areatriggers.
Some code cleanup.

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
fuck nopch and its shit error messages

Commit:
Mage complete

mdX7 commented:
> Warlock, Shaman, and Druid still require some work to make them look like offical.

some work as in: core support or just missing scripts? 
if its missing scripts i'd wait until ur done with all of them until review if you dont mind

malcrom commented:
I have more work to do on the last three classes.

Commit:
Shaman combat complete

Commit:
All working

All classes now working.
Cleaning up code.
There seems to be some gaps in the conversation lines so I will do some more checking so see if some were not sniffed.

malcrom commented:
All classes now working like offical. 
Possibly missing some conversations that were not sniffed so I have to do more research and testing.

malcrom commented:
Issue with Austin spawn not showing until player or follower moves to area he is in for Northbound quest.

https://youtu.be/CMzVXfob3iY

mdX7 commented:
visibility distance in creature addon

malcrom commented:
Nope tried them all. May be phase conditons.

mdX7 commented:
in ur video the npc runs there and you dont change area or satisfy any other possible condition, so in that case i don't think its a phase_area condition issue

malcrom commented:
Fixed the issue

```sql
DELETE FROM `phase_area` WHERE `AreaId`=10452 AND `PhaseId` IN (13846);
INSERT INTO `phase_area` (`AreaId`,`PhaseId`,`Comment`) VALUES
(10452,13846, 'Cosmetic - NPE - Austin Huxworth scout at Quilboar Briarpatch');
```

Commit:
Northbound Quest

Alliance side working.
Horde side to do along with magic number cleanup.

malcrom commented:
Alliance side now working for Northbound quest.

Commit:
Update zone_exiles_reach.cpp

Commit:
Update zone_exiles_reach.cpp

Commit:
Northbound horde quest working

Just cleanup left to do.

malcrom commented:
Horde side now working for Northbound quest. Just cleanup magic numbers left to do.

Commit:
fix error in SQL

malcrom commented:
can't make npc follow on left of player no matter what angle value I set.

Commit:
Remove some magic numbers

Commit:
Fixed follow angle and cleanup

follow at 6 O'clock is Pi, 9 O'clock is Pi/4.

malcrom commented:
Follow fixed.

acidmanifesto commented:
Just spamming here so i can keep following it in the git app

Commit:
Warrior training was coded wrong.

Also did some code changes

malcrom commented:
Complete

Naddley commented:
https://github.com/TrinityCore/TrinityCore/issues/29223

Ovahlord commented on src/server/scripts/ExilesReach/zone_exiles_reach.cpp:
```diff
@@ -35,6 +36,7 @@
 #include "TemporarySummon.h"
 #include "Transport.h"
 #include "Loot.h"
+#include <SpellHistory.h>
```
"SpellHistory.h" pls

malcrom replied:
I don't know what you mean?

malcrom replied:
I need that include for the following:
```cpp
player->GetSpellHistory()->ResetCharges(CARGE_CATAGORY_ID_CHARGE_SPELL);
```

mdX7 replied:
Using quotes "" instead of <>

malcrom replied:
Wow.

Ovahlord commented on src/server/scripts/ExilesReach/zone_exiles_reach.cpp:
```diff
@@ -2186,6 +2188,1317 @@ struct areatrigger_find_the_lost_expedition_follower : AreaTriggerAI
+        }
+        else
+        {
+            _allianceGUID = ObjectGuid::Empty;
```
can be removed. ObjectGuids are default initialized as empty and since this is the very first action that triggers in this script, this operation is not needed. same to _hordeGuid  above this one here

malcrom replied:
Will test

Ovahlord commented on src/server/scripts/ExilesReach/zone_exiles_reach.cpp:
```diff
@@ -2186,6 +2188,1317 @@ struct areatrigger_find_the_lost_expedition_follower : AreaTriggerAI
+        if (me->GetHealth() <= damage)
+        {
+            damage = 0;
+            me->SetHealth(1);
```
damage = me->GetHealth - 1;

This makes the SetHealth hack unneeded and negates any damage beyond the 1 hp threshold

malcrom replied:
yeah i'll remove that

Commit:
Update code

Ovahlord commented:
Looks fine to me :) good job

Commit:
Update zone_exiles_reach.cpp

Commit:
fix error

Commit:
Add phase conditions

Forgot to add phase conditions when I made NPCs viewable from abandoned camp area.

malcrom commented:
Seems to be some phase issues on Northbond again. testing

Commit:
Update conditions to make phasing work

Commit:
Fixed issue with Northbound quest

Fixed issue with Follower would not completing it's script and despawning if you logged off and back on with out reach areatrigger.

malcrom commented:
All issues should be addressed now.

mdX7 committed:
refactor

mdX7 commented:
hunter implementation is missing, could you add it before i merge?

mdX7 committed:
fix whitespace

mdX7 committed:
fix build

mdX7 committed:
fix nopch

malcrom commented:
There is no hunter training. Hunters do a pet quest.

malcrom commented:
You changed the code?

mdX7 committed:
drop hunter

Commit:
Remove TODOs

mdX7 committed:
prepare merge
